### PR TITLE
[Bug](storage )fix dead lock when create_tablet need lock two tablet && update mv_p0…

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
1. fix dead lock when create_tablet need lock two tablet, introduced by #21915
2. update mv_p0/ssb cases
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.66 seconds
 stream load tsv:          537 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17170746653 Bytes

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52.97 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          38 seconds loaded 861443392 Bytes, about 21 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17170214945 Bytes

File: be/src/olap/tablet_manager.cpp [hunk 1]
```diff
@@ -244,7 +244,18 @@ Status TabletManager::create_tablet(const TCreateTabletReq& request, std::vector
...
-    std::lock_guard<std::shared_mutex> wrlock(_get_tablets_shard_lock(tablet_id));
+    // when we create rollup tablet A(assume on shard-1) from tablet B(assume on shard-2)
+    // we need use write lock on shard-1 and then use read lock on shard-2
+    // if there have create rollup tablet C(assume on shard-2) from tablet D(assume on shard-1) at the same time, we will meet deadlock
+    std::unique_lock two_tablet_lock(_two_tablet_mtx, std::defer_lock);
+    bool is_schema_change = request.__isset.base_tablet_id && request.base_tablet_id > 0;
+    bool need_two_lock = is_schema_change && ((_tablets_shards_mask & request.base_tablet_id) !=
+                                              (_tablets_shards_mask & tablet_id));
+    if (need_two_lock) {
+        two_tablet_lock.lock();
```
Comment (yiguolei) [inline be/src/olap/tablet_manager.cpp:255]: do not use this lock, maybe forget to unlock or throw exception before unlock.
maybe use lock guard  + defer lock.

Comment (yiguolei) [inline be/src/olap/tablet_manager.cpp:255]: 这样会导致并发度很低，tablet map lock 分shard 就没啥意义了。 相当于建rollup 变成顺序的了

**Review by yiguolei — APPROVED** [review]
LGTM

File: be/src/olap/tablet_manager.cpp (hunk 1 above)
Comment (yujun777) [reply be/src/olap/tablet_manager.cpp:255]: remove the two_tablet_lock,   lock the two shard sequentially  with their tablet id order ?

Comment (BiteTheDDDDt) [reply be/src/olap/tablet_manager.cpp:255]: > remove the two_tablet_lock, lock the two shard sequentially with their tablet id order ?

The locking time of two_lock is very short, so it has little effect

**Review by yujun777 — APPROVED** [review]
LGTM

**Pull request merged**